diff --git a/dist/server/remark-plugins/remark-headings.js b/dist/server/remark-plugins/remark-headings.js
index 39f387e748c0e200a62b431c5a4cb3a695e112e5..87bd56325c911fc1e56bc02cf99bfb2969bebaef 100644
--- a/dist/server/remark-plugins/remark-headings.js
+++ b/dist/server/remark-plugins/remark-headings.js
@@ -28,7 +28,7 @@ const remarkHeadings = ({ exportName = "toc", isRemoteContent }) => {
           node.data ||= {};
           const headingProps = node.data.hProperties ||= {};
           const value = getFlattenedValue(node);
-          const id = slugger.slug(headingProps.id || value);
+          const id = headingProps.id || slugger.slug(value);
           headingProps.id = id;
           headings.push({ depth: node.depth, value, id });
           return;
